queue.c

```diff
@@ -369,8 +375,7 @@ BaseType_t xQueueGenericReset( QueueHandle_t xQueue,
         }
         else
         {
-            traceQUEUE_CREATE_FAILED( ucQueueType );
-            mtCOVERAGE_TEST_MARKER();
+            configASSERT( pdFAIL );

```

Comment:
This `configASSERT` would not help while debugging. In the previous source, one could simply see which `configASSERT` (with a proper implementation using `__FILE__` and `__LINE__`) was failing and find the issue.
Now one would need to print out the values or step through the code with debugger. Shall we try to find a better way?

Reply:
This was the approach Richard suggested.  Changing the asserts to conditionals means there is still protection when assertions are compiled out, but loses debugging information as you point out.  Adding in the trailing assert is a compromise to provide some information without repeating every check.

Reply:
I'm assuming you mean that we could have an else block where the tests are repeated as asserts.  Manually keeping the two sets of tests in sync strikes me as a big drawback.  Let's get Richard's opinion.

Reply:
This is a great place for an ASSERT_CONTINUE macro. The idea would be to log the assertion to NV storage or or via a debugging port, but not go into the typical infinite loop. This also lets customers test the edge cases in their applications without entering an infinite loop.

Reply:
Same comment as in the queue reset function about using a constant as the parameter in the configASSERT() macro - will generate compiler warnings ("condition is always true").  Extending Paul's thoughts above, what we really need is an assert always function.  I think, also as per my comments in the queue reset function, using a status variable that gets set to pass or fail, then asserting on the status variable just before exiting the function would be fine.  I understand that loses information when debugging, but if you see that line is failing I think it would be possible to see why from within the debugger.

Reply:
changed to assert on the return value